targets/photos/services/onPhotoUpload.js

```diff
@@ -29,93 +28,112 @@ import {
     return
   }
+  const reachs = reachabilities(dataset, spatioTemporalScaled, params)
+  const clusters = gradientClustering(dataset, reachs, params)
+  await saveClustering(clusters)
+}
 
-  const albums = await findAutoAlbums()
-  await computeClusters(dataset, albums, params)
+// Clusterize the given photos, i.e. organize them depending on metrics
+const clusterizePhotos = async (setting, dataset) => {
+  log('info', `Start clustering on ${dataset.length} photos`)
 
-  // TODO save params
+  const albums = await findAutoAlbums()
+  if (albums && albums.length > 0) {
+    await createNewClusters(setting, dataset, albums)
+  } else {
+    // No album found: this is an initialization
+    await createInitialClusters(setting, dataset)
+  }
+  // TODO update params
   // TODO adapt percentiles for large datasets
+  // TODO ensure auto albums unicity
 }
 
-const getNewPhotos = async setting => {
-  const lastSeq = setting.lastSeq
-
+const getNewPhotos = async lastSeq => {
   log('info', `Get changes on files since ${lastSeq}`)
   const result = await cozyClient.fetchJSON(
     'GET',
     `/data/${DOCTYPE_FILES}/_changes?include_docs=true&since=${lastSeq}`
   )
-
   return result.results.map(res => res.doc).filter(doc => {
     return doc.class === 'image' && !doc._id.includes('_design') && !doc.trashed
   })
 }
 
+const initParameters = dataset => {
+  log('info', `Compute clustering parameters on ${dataset.length} photos`)
+  const epsTemporal = computeEpsTemporal(dataset, PERCENTILE)
+  const epsSpatial = computeEpsSpatial(dataset, PERCENTILE)
+  const params = {
+    period: {
+      start: dataset[0].datetime,
+      end: dataset[dataset.length - 1].datetime
+    },
+    modes: [
+      {
+        name: DEFAULT_MODE,
+        epsTemporal: epsTemporal,
+        epsSpatial: epsSpatial
+      }
+    ]
+  }
+  return params
+}
+
 const onPhotoUpload = async () => {
   let setting = await readSetting()
   if (!setting) {
-    setting = await createDefaultSetting()
-  }
-  const photos = await getNewPhotos(setting)
-  if (photos.length > 0) {
-    log('info', `Start clustering on ${photos.length} photos`)
-
-    clusterizePhotos(setting, photos)
+    // No settings found: init them or use default
+    const photos = await getNewPhotos(0)
+    if (photos.length < 1) {
+      log('warn', 'Service called but no photos found to clusterize')
+      return
+    }
+    const dataset = prepareDataset(photos)
+    if (dataset.length > EVALUATION_THRESHOLD) {
+      // There are enough photos to init the parameters and save them
+      const params = initParameters(dataset)
+      setting = await createSetting(params)
+    } else {
+      // Use default
+      setting = defaultSetting(dataset)
+    }
+    await clusterizePhotos(setting, dataset)
+  } else {
+    const photos = await getNewPhotos(setting.lastSeq)
+    if (photos.length < 1) {
+      log('warn', 'Service called but no photos found to clusterize')
+      return
+    }
```

Comment:
Not a big deal, but we could probably factorize this part so we only have to do the check and warning once.